src/hott/01-paths.rzk.md

```diff
@@ -952,3 +964,329 @@ The following is the same as above but with alternating arguments.
+            ( A)
+            ( v)
+            ( v)
+            ( z)
+            ( concat A v w v p q)
+            ( refl )
+            ( H')
+            ( r) ))
+        ( concat
+          ( v = z)
+          ( concat A v v z (concat A v w v p q) r)
+          ( concat A v w z p (concat A w v z q r))
+          ( concat A v w z p (concat A w y z s t))
+          ( associative-concat A v w v z p q r)
+          ( concat-eq-right
+            ( A)
+            ( v)
+            ( w)
+            ( z)
+            ( p)
+            ( concat A w v z q r)
+            ( concat A w y z s t)
+            ( H)))))
+```
+
+It is also convenient to have a a version with the opposite associativity.
+
+```rzk
+#def eq-top-cancel-commutative-square'
+  ( A : U)
+  ( v w y z : A)
+  ( p : v = w)
+  ( q : w = v)
+  ( s : w = y)
+  ( r : v = z)
+  ( t : y = z)
+  ( H : (concat A w v z q r) = (concat A w y z s t))
+  ( H' : (concat A v w v p q) = refl)
+  : r = ( concat A v y z (concat A v w y p s) t)
+  :=
+    (concat
+      ( v = z)
+      ( r)
+      ( concat A v w z p (concat A w y z s t))
+      ( concat A v y z (concat A v w y p s) t)
+      ( concat
+        ( v = z)
+        ( r )
+        ( concat A v v z refl r)
+        ( concat A v w z p (concat A w y z s t) )
+        ( rev
+          (v = z)
+          (concat  A v v z refl r )
+          ( r )
+          (left-unit-concat A v z r))
+        ( concat
+          ( v = z)
+          ( concat A v v z refl r)
+          ( concat A v v z (concat A v w v p q) r )
+          ( concat A v w z p (concat A w y z s t ))
+          ( rev
+            ( v = z )
+            ( concat A v v z  (concat A v w v p q) r )
+            ( concat A v v z refl r)
+            ( concat-eq-left
+              ( A)
+              ( v)
+              ( v)
+              ( z)
+              ( concat A v w v p q )
+              ( refl )
+              ( H')
+              ( r) ))
+          ( concat
+            ( v = z)
+            ( concat A v v z (concat A v w v p q) r)
+            ( concat A v w z p (concat A w v z q r)  )
+            ( concat A v w z p (concat A w y z s t))
+            ( associative-concat A v w v z p q r)
+            ( concat-eq-right
+              ( A )
+              ( v )
+              ( w)
+              ( z)
+              ( p)
+              ( concat A w v z q r)
+              ( concat A w y z s t)
+              ( H)))))
+      ( rev-associative-concat A v w y z p s t))
+```
+
+And a reversed version.
+
+```rzk
+#def rev-eq-top-cancel-commutative-square'
+  ( A : U)
+  ( v w y z : A)
+  ( p : v = w)
+  ( q : w = v)
+  ( s : w = y)
+  ( r : v = z)
+  ( t : y = z)
+  ( H : (concat A w v z q r) = (concat A w y z s t))
+  ( H' : (concat A v w v p q) = refl)
+  : ( concat A v y z (concat A v w y p s) t) = r
+  :=
+    rev
+      ( v = z)
+      ( r)
+      ( concat A v y z (concat A v w y p s) t)
+      ( eq-top-cancel-commutative-square' A v w y z p q s r t H H')
```

Comment:
We're trying out a new indentation scheme for the top level part of a definition, to save on horizontal space:
```suggestion
  :=
  rev
  ( v = z)
  ( r)
  ( concat A v y z (concat A v w y p s) t)
  ( eq-top-cancel-commutative-square' A v w y z p q s r t H H')
```